spec/require_encapsulation_check.rb

```diff
@@ -0,0 +1,31 @@
+require 'English'
+REQUIRES = %w[
+  datadog/appsec
+  datadog/core
+  datadog/kit
+  datadog/profiling
+  datadog/tracing
+].freeze
+
+RSpec.describe 'require encapsulation' do
```

Comment:
In a way I feel this description isn't quite correct, which led me to think that this spec isn't quite how we expect it to be.

Notably it puts some cross-component knowledge into that spec file. I think it might be better if each "require "works is isolated in each separate component:

- assert that `require datadog/core` works => in `spec/datadog/core_spec.rb`
- assert that `require datadog/appsec` works => in `spec/datadog/appsec_spec.rb`
- ... and so on